src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultConverters.cs

```diff
@@ -2,12 +2,27 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections.Concurrent;
 using System.Reflection;
 
 namespace System.Text.Json.Serialization.Converters
 {
     internal static class DefaultConverters
     {
+        private static readonly ConcurrentDictionary<Type, object> _valueConverters = new ConcurrentDictionary<Type, object>();
+
+        static DefaultConverters()
+        {
+            _valueConverters.TryAdd(typeof(DateTimeOffset), new JsonValueConverterDateTimeOffset());
```

Comment:
Do we need to add `typeof(DateTime)` as well?

Reply:
No, because oddly enough `DateTime` has a `TypeCode` and is handled below.

Reply:
Can you please add a comment to that effect?